src/main.cpp

```diff
@@ -76,8 +78,21 @@ class MetaGenerationConsumer : public clang::ASTConsumer {
         // Generate TypeScript definitions
         if (!cla_outputDtsFolder.empty()) {
             llvm::sys::fs::create_directories(cla_outputDtsFolder);
+
+            // find iOS SDK docset path
+            std::string docSetPath;
+            if (!cla_docSetFile.empty()) {
+                docSetPath = cla_docSetFile.getValue();
+            }
+            else if (struct passwd* pwd = getpwuid(getuid())) {
```

Comment:
I think it would be best to not try to compute the docset path inside the metadata generator. I think it would be better to do it in the python script that drives the metadata generator, because it can access the EFFECTIVE_PLATFORM_NAME build setting in Xcode in order to find the docset for the right platform.